cdk/infrastructure/build.py

```diff
@@ -21,10 +21,16 @@ class Args:
     config_name: str
 
 
+def branch_name_is_too_long(branch_name: str) -> bool:
+    return len(branch_name) > 44
+
+
 def get_args(app: App) -> Args:
     branch = app.node.try_get_context('branch')
     if branch is None:
         raise ValueError('Must specify branch context: `-c branch=$BRANCH`')
+    if branch_name_is_too_long(branch):
+        raise ValueError('Branch name must not be longer than 44 characters.')
```

Comment:
Maybe should put actual length in error?

`f'Branch name too long ({len(branch} > 44)`?